pkg/collectors/virtualservices.go

```diff
@@ -0,0 +1,110 @@
+package collectors
+
+import (
+	"context"
+	"fmt"
+	"github.com/douglas-reid/istio-state-metrics/pkg/apis/networking/v1alpha3"
+	"github.com/douglas-reid/istio-state-metrics/pkg/client/clientset/versioned"
+	informers "github.com/douglas-reid/istio-state-metrics/pkg/client/informers/externalversions/networking/v1alpha3"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/client-go/tools/cache"
+)
+
+var (
+	descVirtualServiceHostsName          = "istio_pilot_virtual_service_hosts"
```

Comment:
question: do you think it is better to have this be a 1:1 mapping?

something like:
```
istio_pilot_virtual_service_host_info{
  virtual_service = "bookinfo",
  namespace = "default",
  host = "productpage",
}
```

That way you could easily go from host to virtual service, as well as counting number of hosts per virtual service.

Same question for gateways.

Reply:
I think it will be a 1:1 mapping as follows:

istio_pilot_virtual_service_hosts_info{
  virtual_service = "bookinfo.default",
  hosts = "productpage",
}

I think having hosts in the name is confusing.. will change it to host and same for gateways